### PR TITLE
DOC added titles to the rest of the 3D plots

### PR DESCRIPTION
**Pull request by NelleV**
This PR should finish up the migration of the axes3D plots to sphinx-gallery. The titles aren't great, but this should help finish up the migration quickly.

### CODE REVIEW

File: examples/mplot3d/lines3d_demo.py [hunk 1]
```diff
@@ -1,4 +1,8 @@
 '''
+=========================
+3D parametric curve in 3D
```
Comment (WeatherGod) [inline examples/mplot3d/lines3d_demo.py]: "in 3D" is a bit redundant.

File: examples/mplot3d/mixed_subplots_demo.py [hunk 2]
```diff
@@ -1,4 +1,8 @@
 """
+=================
+2D and 3D suplots
```
Comment (WeatherGod) [inline examples/mplot3d/mixed_subplots_demo.py]: typo: "suplots" --> "subplots"

**Comment by NelleV** [comment]
Thanks for the review @WeatherGod . I've fixed the two titles.

**Comment by phobson** [comment]
Appveyor is definitely broken.
Travis failure looks spurious, but I'll restart the one failing travis build just to be sure.

File: examples/mplot3d/contour3d_demo.py [hunk 3]
```diff
@@ -1,5 +1,7 @@
 '''
+===================================================
 Demonstrates plotting contour (level) curves in 3D.
```
Comment (QuLogic) [inline examples/mplot3d/contour3d_demo.py]: Remove period? (was done on the next one)

File: examples/mplot3d/custom_shaded_3d_surface.py [hunk 4]
```diff
@@ -1,4 +1,9 @@
 """
+=======================================
+Custom hillshading in a 3D surface plot
+=======================================
+
+
```
Comment (QuLogic) [inline examples/mplot3d/custom_shaded_3d_surface.py]: Extra blank line.

**Pull request merged**